src/main/java/uk/gov/digital/ho/hocs/casework/migration/api/MigrationCaseDataService.java

```diff
@@ -0,0 +1,81 @@
+package uk.gov.digital.ho.hocs.casework.migration.api;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import uk.gov.digital.ho.hocs.casework.api.dto.CaseDataType;
+import uk.gov.digital.ho.hocs.casework.client.auditclient.AuditClient;
+import uk.gov.digital.ho.hocs.casework.client.infoclient.InfoClient;
+import uk.gov.digital.ho.hocs.casework.domain.exception.ApplicationExceptions;
+import uk.gov.digital.ho.hocs.casework.domain.model.CaseData;
+import uk.gov.digital.ho.hocs.casework.domain.repository.CaseDataRepository;
+
+import java.time.LocalDate;
+import java.util.Map;
+import java.util.UUID;
+
+import static net.logstash.logback.argument.StructuredArguments.value;
+import static uk.gov.digital.ho.hocs.casework.application.LogEvent.*;
+import static uk.gov.digital.ho.hocs.casework.client.auditclient.EventType.CASE_UPDATED;
+
+@Service
+@Slf4j
+@Qualifier("MigrationCaseDataService")
+public class MigrationCaseDataService {
+    protected final CaseDataRepository caseDataRepository;
+    protected final AuditClient auditClient;
+    protected final InfoClient infoClient;
+
+    public MigrationCaseDataService(CaseDataRepository caseDataRepository,
+                                    InfoClient infoClient,
+                                    AuditClient auditClient) {
+        this.caseDataRepository = caseDataRepository;
+        this.infoClient = infoClient;
+        this.auditClient = auditClient;
+    }
+
+    protected CaseData getCaseData(UUID caseUUID) {
+        log.debug("Getting Case: {}", caseUUID);
+        CaseData caseData = caseDataRepository.findActiveByUuid(caseUUID);
+        if (caseData == null) {
+            log.error("Case: {}, not found!", caseUUID, value(EVENT, CASE_NOT_FOUND));
```

Comment:
Should we have `MIGRATION_CASE_NOT_FOUND` or is using the defaults okay for these events?

Reply:
Added